src/transformers/models/dpt/modeling_flax_dpt.py

```diff
@@ -0,0 +1,1237 @@
+            The data type of the computation. Can be one of `jax.numpy.float32`, `jax.numpy.float16` (on GPUs) and
+            `jax.numpy.bfloat16` (on TPUs). This can be used to enable mixed-precision training or half-precision
+            inference on GPUs or TPUs. If specified all the computation will be performed with the given `dtype`.
+            **Note that this only specifies the dtype of the computation and does not influence the dtype of model
+            parameters.** If you wish to change the dtype of the model parameters, see [`~FlaxPreTrainedModel.to_fp16`]
+            and [`~FlaxPreTrainedModel.to_bf16`].
+"""
+
+DPT_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`numpy.ndarray` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`DPTFeatureExtractor`]. See
+            [`DPTFeatureExtractor.__call__`] for details.
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+# Copied from transformers.models.vit.modeling_flax_vit.FlaxPatchEmbeddings with ViT->DPT
+class FlaxPatchEmbeddings(nn.Module):
+
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32  # the dtype of the computation
+
+    def setup(self):
+        image_size = self.config.image_size
+        patch_size = self.config.patch_size
+        num_patches = (image_size // patch_size) * (image_size // patch_size)
+        self.num_patches = num_patches
+        self.projection = nn.Conv(
+            self.config.hidden_size,
+            kernel_size=(patch_size, patch_size),
+            strides=(patch_size, patch_size),
+            padding="VALID",
+            dtype=self.dtype,
+            kernel_init=jax.nn.initializers.normal(self.config.initializer_range),
+        )
+
+    def __call__(self, pixel_values):
+        x = self.projection(pixel_values)
+        batch_size, _, _, channels = x.shape
+        return jnp.reshape(x, (batch_size, -1, channels))
+
+
+# Copied from transformers.models.vit.modeling_flax_vit.FlaxViTEmbeddings with ViT->DPT
+class FlaxDPTEmbeddings(nn.Module):
+    """Construct the CLS token, position and patch embeddings."""
+
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32  # the dtype of the computation
+
+    def setup(self):
+        self.cls_token = self.param("cls_token", nn.initializers.zeros, (1, 1, self.config.hidden_size))
+        self.patch_embeddings = FlaxPatchEmbeddings(self.config, dtype=self.dtype)
```

Comment:
@NielsRogge I think that in this implementation we directly initialize the modules with the config (contrary than in `DPTViTEmbeddings` if this is what you meant?

Reply:
I copied these modules from `FlaxViTModel` which seem to have the right structure as suggested in #17731

Reply:
Ok, then it's alright :)

Reply:
Oh but the sanity check of the channel size is missing indeed, will add that!

Reply:
Yeah we don't have it for other Flax models as well right now. Ideally (and also for consistency), we should have it for all vision models